src/share/content_renderer.js

```diff
@@ -57,21 +57,28 @@ function getContent(note) {
             }
 
             content = document.body.innerHTML;
+ 
+            if (content.includes(`<span class="math-tex">`)) {
+                content += `<script src="../../libraries/katex/katex.min.js"></script>`;
+                content += `<link rel="stylesheet" href="../../libraries/katex/katex.min.css">`;
+                content += `<script src="../../libraries/katex/auto-render.min.js" onload="renderMathInElement(document.getElementById('content'));"></script>`;
+                content += `<script src="../../libraries/katex/mhchem.min.js"></script>`;
+            }
         }
     }
-    else if (note.type === 'code' || note.type === 'mermaid') {
+    else if (note.type === 'code') {
         if (!content?.trim()) {
-            content = NO_CONTENT + getChildrenList(note);
+            content = NO_CONTENT;
         }
         else {
-            const document = new JSDOM().window.document;
-
-            const preEl = document.createElement('pre');
-            preEl.appendChild(document.createTextNode(content));
-
-            content = preEl.outerHTML;
+            content = `<textarea style="width:100px;" id="code">${content}</textarea>`
+            content += `<link rel="stylesheet" href="../../libraries/codemirror/codemirror.css">`
+            content += `<script src="../../libraries/codemirror/codemirror.js" onload="var editor = CodeMirror.fromTextArea(document.getElementById('code'), {lineNumbers: true, lineWrapping: true});"></script>`
```

Comment:
I could parse the mime type to edit the codemirror language for better formatting. Not sure how I would do that but `MIME_TYPES_DICT` seems helpful